File: 성민/다시풀문제/소수_찾기.js
//에라토스 테네스체 알고리즘
function solution(n) {
  let answer = 0;
  const arr = new Array(n + 1).fill(true); // 초깃값 설정
  const end = Math.sqrt(n);

  for (let i = 2; i <= end; ++i) {
    // 이미 소수가 아닌 인덱스는 건너뛴다.
    if (arr[i] === false) {
      continue;
    }
    // 소수가 아닌 데이터는 false로 입력한다.
    for (let k = i * i; k <= n; k += i) {
      arr[k] = false;
    }
  }
  // 소수의 갯수를 구한다.
  for (let i = 2; i <= n; ++i) {
    if (arr[i] === true) {
      answer++;
    }
  }
  return answer;
}

solution(50);

File: 종우/프로그래머스/소수찾기.py
# 0207
# https://school.programmers.co.kr/learn/courses/30/lessons/42839#

import itertools
import math

def solution(numbers):
    primes = set()
    for i in range(len(numbers)):
        p = itertools.permutations(numbers, i+1)
        for j in p:
            num = int("".join(j))
            if isprime(num):
                primes.add(num)
    return len(primes)

def isprime(n):
    if n < 2:
        return False
    else:
        for i in range(2, int(math.sqrt(n))+1):
            if n%i == 0:
                return False
        return True